storage/innobase/include/buf0buf.h

```diff
@@ -648,10 +648,10 @@ class buf_page_t
   void init(uint32_t state, page_id_t id)
   {
     ut_ad(state < REMOVE_HASH || state >= UNFIXED);
+    ut_ad(!lock.is_locked_or_waiting());
     id_= id;
     zip.fix= state;
     oldest_modification_= 0;
-    lock.init();
```

Comment:
I see this lock.init() call is moved to the caller. Can we assert here that the lock is initialized ?

Reply:
The added assertion checks a part of the effect of `lock.init()`. It will not cover the debug-only field `readers_lock` or `readers` for keeping track of threads that are holding S latches, nor the `pthread_mutex_t` or `pthread_cond_t` that would be embedded in `ssux_lock_impl` when `SUX_LOCK_GENERIC` is defined (`futex`-like system calls are not available).

However, other checks about the initialization or destruction of the debug field `readers_lock` would easily fire when `SUX_LOCK_GENERIC` is defined and the initialization was forgotten. The reason why the `lock.init()` call was moved out of here was to ensure that all `init()` and `destroy()` calls are paired.